### DISCUSSION TITLE
Ugly shadows on menus in Chrome 119.0

### DISCUSSION THREAD

**Discussion by Filip-K** [regolith-linux/regolith-desktop#949] [Q&A] [answered]
Hi! I started experiencing the following picom shadow glitch after a recent Chrome update. Some menus (the three-dot, the right-click) have ugly transparent frames with shadows:
![chrome_ugly_2](https://github.com/regolith-linux/regolith-desktop/assets/16614296/deb6c520-60c5-482f-a1cb-c34445c30a61)
![chrome_ugly_1](https://github.com/regolith-linux/regolith-desktop/assets/16614296/bc9ea761-c0d4-4397-9af2-ecf469ddf2bf)

Interestingly, some menus are fine:
![chrome_ugly_3](https://github.com/regolith-linux/regolith-desktop/assets/16614296/ce9f1d3b-a256-4023-8ab1-7139db6c0458)

@kgilmer I remember we removed some shadow exceptions for Chrome in the picom config a while back, because they didn't seem necessary.  Maybe it's time to bring them back? 😬 Anybody else experiencing the same issue?

**Comment by diffficult** [comment]
what shadow exceptions were those?

**Reply by Filip-K** [reply]
I can't remember and I couldn't find the right commit. I'm not even sure in which repo to look, as the picom config apparently hopped between repos a few times. In either case, these exceptions were (from what I remember) disabling shadows for all Chrome windows, so even if they would solve this issue, it wouldn't be a good long term solution.

**Comment by aav** [comment]
can confirm the same issue:

![image](https://github.com/regolith-linux/regolith-desktop/assets/69563/49240c7a-1323-451a-905f-8110ff812076)

**Comment by jq-compose** [comment]
For others that hit this discussion from a search engine, adding a compositor (e.g. picom) may fix this. I'm seeing the same issue in NixOS / i3 window manager (with no compositor). 

Reddit thread with the same problem described:

https://www.reddit.com/r/linuxquestions/comments/17sq1hv/how_to_disable_chrome_context_menu_border/

Quote from [a comment](https://www.reddit.com/r/linuxquestions/comments/17sq1hv/comment/kbl62cm/) in that thread with a solution for Debian:

> I had the same problem on Debian 11 KDE and i3-wm.
> 
> I solved it by installing picom
> 
> (sudo apt-get -y install picom)
> 
> and adding this line to the "~/.config/i3/config" file:
> 
> exec --no-startup-id picom
> 
> to start it automatically
> 

For NixOS, some options may be:

https://rycee.gitlab.io/home-manager/options.html#opt-services.picom.enable

Edit: I can confirm that using `picom` on NixOS solves my problem. After re-reading this thread, it's probably not related to this specific problem though. Sorry for the noise and getting off topic here.

**Reply by Filip-K** [reply]
Thanks for chiming in! This is related but not exactly the same issue. Running without picom gives a solid-color frame
![no-picom](https://github.com/regolith-linux/regolith-desktop/assets/16614296/e9658dbc-739a-44c6-8d9a-2fe715aafdab)
And adding picom (but disabling shadows) solves it
![with-picom-without-shadows](https://github.com/regolith-linux/regolith-desktop/assets/16614296/4311f263-ac99-4225-a172-24c4506ab149)
But if you run picom with shadows (the default in regolith), you get this and that's mostly what the issue is about
![with-picom](https://github.com/regolith-linux/regolith-desktop/assets/16614296/ca8fb7df-bd7c-4ba3-bc9a-f448066a8f09)

**Comment by teaper** [comment, marked as answer]
在 `/etc/xdg/picom.conf` 中设置 `menu = { shadow = false; }` 可以解决

```bash
wintypes:
{
  tooltip = { fade = true; shadow = true; opacity = 0.75; focus = true; full-shadow = false; };
  dock = { shadow = false; clip-shadow-above = true; }
  dnd = { shadow = false; }
  menu        = { shadow = false; }
  popup_menu = { opacity = 0.8; }
  dropdown_menu = { opacity = 0.8; }
};

**Reply by Filip-K** [reply]
Spot on, thanks @teaper!

**Reply by p3k** [reply]
The suggestion in the answer does not work for me. Checking the processes I found that picom is run  with `--config /etc/regolith/picom/config`.

If I set `shadow = false;` in `/etc/regolith/picom/config` the shadows of the Chrome menus disappear.

But I assume that setting is prone to being overwritten…? What is the right way to fix this?

**Reply by p3k** [reply]
Ah found this [in the docs](https://regolith-desktop.com/docs/howtos/customize-compositor/#overriding-regolith-default-compositor-configuration):

> To provide your own Compton/Picom compositor config, copy the default or create your own and save it as ~/.config/regolith3/picom/config

So at least I can maintain a full configuration without risk of being overwritten.

**Reply by p3k** [reply]
Still struggling to exclude Chrome browser with the `shadow-exclude` setting…

I tried these settings and some variations of it:

```plain
shadow-exclude = [
  "name *?= 'chrome'";
  "class_i *?= 'chrome'";
  "_NET_WM_NAME@:s *?= 'chrome'";
];
```

(See https://man.archlinux.org/man/picom.1)

Any more suggestions?

**Reply by theFahmid** [reply]
I am using Wayland. Will the same thing apply?

**Reply by kgilmer** [reply]
@fahmidbinfarooqui , no in this case `picom` is used only by the X11 Regolith session.  The wayland session does not utilize this type of component.

**Reply by kgilmer** [reply]
@teaper & @Filip-K and others interested in this change , would you consider these picom config changes to be generally useful?  I opened a PR for comments: https://github.com/regolith-linux/regolith-compositor-picom-glx/pull/11

**Reply by theFahmid** [reply]
> @fahmidbinfarooqui , no in this case `picom` is used only by the X11 Regolith session.  The wayland session does not utilize this type of component. 

I have the same problem in Hyprland. Can you tell me what to do?

**Comment by waldo2810** [comment]
This is working in picom vgit-b700a:

```
wintypes:
{
  menu = { shadow = false; blur-background = false;}
};
```

**Reply by kbc8090** [reply]
yea blur-background = false was it
